### ISSUE TITLE
Splitting the road does also cause wrong order of a public transport version 2 relation's members

### ISSUE THREAD

**Issue by Lukas458** [openstreetmap/iD#4864]
If you split a way in iD, which is part of a public transport version 2 relation, the relation's members with the roles "stop" and "platform" are ordered completely new after splitting the way. Even if the members with the roles "stop" and "platform" have nothing to do with the way splitted in two ones. First are listed all "stop" members, then all "platform" members. For example, please look at this relation: https://www.openstreetmap.org/relation/1732029 The order of the members is wrong. After every stop member, there has to come the matching platform member, if existent. Not all stop members first and then all platform mebers second.

**Comment by bhousel** [comment]
Thanks for flagging @Lukas458, I did not realize that the nodes were being reordered.  I'll see if this is something we can support in iD.

Relevant code that does this would be _around_ here somewhere: https://github.com/openstreetmap/iD/blob/a01ddf26f9bf0ed5eaf1191fccf4e45b7ebfd4a9/modules/actions/add_member.js#L119-L122
My expectation was that `groups.node` would contain all the nodes from the Relation in the order that they existed before we do any way reordering. 

for reference:  https://wiki.openstreetmap.org/wiki/Public_transport#Service_routes

**Comment by slhh** [comment]
@Lukas458 
Are the stops still in correct sequence?

@bhousel All platforms are ways in this route, which is pretty valid in PTv2. Maybe, this is the reason for the issue.

> // Write members in the order: nodes, ways, relations 
This is wrong, at least for PTv2.

The wiki says:
>  first a list of all stations in the order: stop_position 1, platform 1, stop_position 2, platform 2, … Stop positions get the role stop, platforms the role platform. It is necessary to add the stops/platforms if they are mapped but it is not necessary to add them if they are missing just to be able to add them to the relation.
The list of stops and platforms is followed by a ordered list of the ways which are used by the vehicle from the starting station to the terminal station.

**Comment by bhousel** [comment]
> All platforms are ways in this route, which is pretty valid in PTv2. Maybe, this is the reason for the issue.

Oh no.. yeah that's definitely the issue.
We'll need to exclude members with roles `stop` and `platform` from the way reordering code. 😓

**Comment by Lukas458** [comment]
THere seems to be another, let's call it "problem": Not only PTV2-relation's members with the roles _stop_ and _platform_ need to be **excluded** **from the way reordering code**, **but also members with the roles _stop_entry_only_, _platform_entry_only_, _stop_exit_only_ and _platform_exit_only_**. I know it's a bit complicated, but please could anyone help? :)

**Comment by lakedistrictOSM** [comment]
Not sure if this is a duplicate, related or needs a new issue -  I created a bus route relation in iD (https://www.openstreetmap.org/relation/8191135/history v1) and all stops, platforms and roads were in the correct order apart from the last bus stop and platform which was added to the relation after the road ways.

The expected behaviour is the order of members as they are in v2.
The actual behaviour was the order of members as they were in v1.

Note that this bus stop (Windermere Railway Station (2)) is visited twice on this circular route.
